apps/experiments/filters.py

```diff
@@ -124,17 +127,48 @@ def build_participant_filter(operator, value):
     return None
 
 
-def build_timestamp_filter(operator, value, field=None):
-    """Build filter condition for timestamp"""
+def build_timestamp_filter(operator, value, field=None, timezone=None):
+    """Build filter condition for timestamp, supporting date and relative ranges like '1h', '7d'.
+    For 1d 24h are subtracted i.e sessions in the range of 24h are shown not based on the date"""
+
+    if not value or not field:
+        return None
+
     try:
-        date_value = datetime.strptime(value, "%Y-%m-%d").date()
-        if operator == Operators.ON:
-            return Q(**{f"{field}__date": date_value})
-        elif operator == Operators.BEFORE:
-            return Q(**{f"{field}__date__lt": date_value})
-        elif operator == Operators.AFTER:
-            return Q(**{f"{field}__date__gt": date_value})
-    except (ValueError, TypeError):
+        client_tz = pytz.timezone(timezone) if timezone else pytz.UTC
+        now_client = datetime.now(client_tz)
+        # Handle 'range' operator with relative time (e.g., '1h', '7d')
+        if operator == Operators.RANGE:
+            if value.endswith(("h", "d", "m")):
+                num = int(value[:-1])
+                unit = value[-1]
+
+                if unit == "h":
+                    delta = timedelta(hours=num)
+                elif unit == "d":
+                    delta = timedelta(days=num)
+                elif unit == "m":
+                    delta = timedelta(minutes=num)
+                else:
+                    return None
+
+                range_starting_client_time = now_client - delta
+                range_starting_utc_time = range_starting_client_time.astimezone(pytz.UTC)
+                return Q(**{f"{field}__gte": range_starting_utc_time})
+
+            else:
+                return None
+        else:
+            # No need to convert the date as we are passing only date and date shown on the UI is in client time only
+            date_value = datetime.strptime(value, "%Y-%m-%d").replace(tzinfo=UTC)
+
+            if operator == Operators.ON:
+                return Q(**{f"{field}__date": date_value})
```

Comment:
After converting the user date to a UTC datetime this will need adjusting: `Q(**{f"{field}__date": local_datetime.date()})`